src/acl/Acl.cc

```diff
@@ -143,27 +149,39 @@ void Acl::Node::operator delete(void *)
     fatal ("unusable Acl::Node::delete");
 }
 
-Acl::Node *
-Acl::Node::FindByName(const char *name)
+/// implements both Acl::Node::FindByName() variations
+template <typename SBufOrCString>
+static Acl::Node *
+FindByNameT(const SBufOrCString name)
 {
-    debugs(28, 9, "name=" << name);
-
-    for (auto *a = Config.aclList; a; a = a->next)
-        if (!strcasecmp(a->name, name))
+    for (auto a = Config.aclList; a; a = a->next) {
+        if (a->name.caseCmp(name) == 0) {
+            debugs(28, 8, "found " << a->name);
             return a;
+        }
+    }
 
-    debugs(28, 9, "found no match");
-
+    debugs(28, 8, "cannot find " << name);
     return nullptr;
 }
 
+Acl::Node *
+Acl::Node::FindByName(const SBuf &name)
+{
+    return FindByNameT<const SBuf &>(name);
```

Comment:
Why do we need to specify the template argument here? Can't we just rely on template argument deduction?
```suggestion
    return FindByNameT(name);
```

Reply:
> Why do we need to specify the template argument here?

We do not have to. I think being explicit adds clarity in this case. However, I removed it in hope to make progress with this [head-of-line](https://en.wikipedia.org/wiki/Head-of-line_blocking) PR.

> Can't we just rely on template argument deduction?

We can. Done in commit 01a25ae.